Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/PatientCorrelationServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+        PRPAIN201301UV02 request = HL7PRPA201301Transforms.createPRPA201301(patient, aa, aa, aa);
+        
+        correlationClient.addPatientCorrelation(request, assertion);
+        return newCorrelation;
+    }
+
+    private String getLocalAA() {
+        if (NullChecker.isNullish(localAA)) {
+            try {
+                localAA = getPropertyAccessor().getProperty(NhincConstants.ADAPTER_PROPERTY_FILE_NAME, NhincConstants.ASSIGNING_AUTH_PROPERTY);
+            } catch (PropertyAccessException ex) {
+                LOG.warn("Unable to get local assigning authority due to: {}", ex.getLocalizedMessage());
+            }
+        }
+        return localAA;
+    }
+
+    private PropertyAccessor getPropertyAccessor() {
+        return PropertyAccessor.getInstance();
+    }
+
+    private PRPAIN201309UV02 buildPRPAINRequest(PatientSearchResults patientResults) {
+        RetrievePatientCorrelationsRequestType retrieveCorrelations = new RetrievePatientCorrelationsRequestType();
+
+        if (!patientResults.isPatientListEmpty() && NullChecker.isNotNullish(patientResults.getPatientList().get(0).getPid())
+                && NullChecker.isNotNullish(patientResults.getPatientList().get(0).getAaId())) {
+            QualifiedSubjectIdentifierType qualPatientId = new QualifiedSubjectIdentifierType();
+            qualPatientId.setAssigningAuthorityIdentifier(patientResults.getPatientList().get(0).getAaId());
+            qualPatientId.setSubjectIdentifier(patientResults.getPatientList().get(0).getPid());
+            retrieveCorrelations.setQualifiedPatientIdentifier(qualPatientId);
+            return new PixRetrieveBuilder().createPixRetrieve(retrieveCorrelations);
+        }
+
+        return null;
+    }
+
+    private void addToCorrelationList(RetrievePatientCorrelationsResponseType response, PatientSearchResults patientResults) {
+        if(!patientResults.isPatientListEmpty() && patientResults.getPatientList().get(0) != null
+             && response != null && response.getPRPAIN201310UV02() != null
+             && response.getPRPAIN201310UV02().getControlActProcess() != null
+             && response.getPRPAIN201310UV02().getControlActProcess().getSubject() != null) {
+            List<PRPAIN201310UV02MFMIMT700711UV01Subject1> subjectList = response.getPRPAIN201310UV02().getControlActProcess().getSubject();
+            for (PRPAIN201310UV02MFMIMT700711UV01Subject1 subject : subjectList) {
+                if(subject.getRegistrationEvent() != null && subject.getRegistrationEvent().getSubject1() != null
+                        && subject.getRegistrationEvent().getSubject1().getPatient() != null) {
+                    patientResults.getPatientList().get(0).getCorrelations().addAll(subject.getRegistrationEvent().getSubject1().getPatient().getId());
+                }
+            }
+        }
+    }
+
+    private String generatePatientId() {
+        int number = new Random().nextInt(10000000);
```

Comment:
While *excessively* unlikely there is still the possibility of this producing a conflict. Perhaps we could get the current epoc time in milliseconds and use that number instead? Unless we are able to call this method more than once at the exact same time, it wont produce a collision from a data standpoint.

Reply:
What did you do for generating document ID?

Reply:
For the document ID I queried what was currently in the database and incremented the number. If there was no result in the database, I returned a default value of 0, then concatenated it with "CONNECT" + [ID]

https://github.com/CONNECT-Solution/CONNECT/blob/CONNECT_integration/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docrepository/adapter/dao/DocumentDao.java#L95

Reply:
I don't have access to the correlation database from the admin gui and would have to introduce a new service which seems extremely excessive.  Very, very low likelihood of calling this method at the exact same time as this is really adapter logic through a portal.

Reply:
That is why I was suggesting using the current time on the OS's clock and use that number. You don't need any connection to a DB. the current time will be an ever-increasing number which means you wont ever have any conflicts. Even if you have 2 requests fire off at the exact same time, the likelihood they will get processed in the exact same millisecond is virtually impossible.

 return "CONNECT" + new Date().getTime();

would be all that's needed, and never produce a conflict.

Reply:
Changed.